### PR TITLE
DRILL-6140: Correctly list Operators in a Query Profile

### PR DESCRIPTION
**Pull request by kkhatua**
Operators listed in Profiles Page don't always correspond with operator specified in Physical Plan.
This commit fixes that by using the PhysicalPlan as a reference, but reverts to the inferred names in the event of an Exchange-based operator.
This is trying to address the example in the JIRA, where we need to show up a `FLATTEN` operator correctly, but there is no such operator type defined in `org.apache.drill.exec.proto.UserBitShared.CoreOperatorType`. So, the operator shown instead is `SINGLE_SENDER`

### CODE REVIEW

**Comment by kkhatua** [comment]
@arina-ielchiieva please review

File: exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/profile/OperatorWrapper.java [hunk 1]
```diff
@@ -45,12 +45,25 @@
...
     this.major = major;
     firstProfile = opsAndHostsList.get(0).getLeft().getLeft();
     operatorType = CoreOperatorType.valueOf(firstProfile.getOperatorType());
-    operatorName = operatorType == null ? UNKNOWN_OPERATOR : operatorType.toString();
+    //Update Name from Physical Map
+    String path = new OperatorPathBuilder().setMajor(major).setOperator(firstProfile).build();
+    //Use Plan Extracted Operator Names if available
+    String extractedOpName = phyOperMap.get(path);
+    String inferredOpName = operatorType == null ? UNKNOWN_OPERATOR : operatorType.toString();
+    //Revert to inferred names for exceptional cases
```
Comment (amansinha100) [inline exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/profile/OperatorWrapper.java]: Can you clarify 'exception cases' .. there are 2 examples below but it is not immediately clear which operators would fall into this pattern.  I assume the main mismatch is that the physical plan has EXCHANGE operators whereas the profile has SENDER and RECEIVER.

Comment (kkhatua) [reply exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/profile/OperatorWrapper.java]: Added all the exceptions in comments

File: exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/profile/OperatorWrapper.java [hunk 2]
```diff
@@ -45,12 +45,25 @@
...
+    if (extractedOpName == null || inferredOpName.contains(extractedOpName)
+        || inferredOpName.endsWith("_RECEIVER") || inferredOpName.endsWith("_RECEIVER")) {
```
[the remark this answers is not in the document]
Comment (kkhatua) [reply exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/profile/OperatorWrapper.java]: Fixed it. Refactored to checking for *_EXCHANGE

File: exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/profile/ProfileWrapper.java [hunk 3]
```diff
@@ -322,4 +330,16 @@ public String getOperatorsJSON() {
...
     return onlyImpersonationEnabled;
   }
+
+  //Generates operator names inferred from physical plan
+  private void generateOpMap(String plan) {
+    this.physicalOperatorMap = new HashMap<String,String>();
+    String[] operatorLine = plan.split("\\n");
+    for (String line : operatorLine) {
+      String[] lineToken = line.split("\\s+", 3);
+      String operatorPath = lineToken[0].trim().replaceFirst("-", "-xx-"); //Required format for lookup
```
[the remark this answers is not in the document]
Comment (kkhatua) [reply exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/profile/ProfileWrapper.java]: Added the following as the example
```
01-03 Flatten(flattenField=[$1]) : rowType = RecordType(ANY rfsSpecCode, ...
```

**Comment by amansinha100** [comment]
Updated version lgtm.  +1

**Pull request closed without merging**